src/main/java/jp/co/soramitsu/sora/didresolver/validation/KeyValidator.java

```diff
@@ -0,0 +1,25 @@
+package jp.co.soramitsu.sora.didresolver.validation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import jp.co.soramitsu.sora.crypto.Consts;
+import jp.co.soramitsu.sora.didresolver.validation.constrains.KeyConstraint;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class KeyValidator implements ConstraintValidator<KeyConstraint, String> {
+
+  private static final String KEY_FORMAT_PATTERN = ".*" + Consts.DID_URI_DETERMINATOR + "keys-\\d";
+
+  @Override
+  public boolean isValid(String publicKey, ConstraintValidatorContext context) {
+    log.debug("validation format of public key - {}", publicKey);
```

Comment:
did is uri

why you don't use java.net.URI? it has .getFragment or something like that

Reply:
yes, but for DID validation you still have to parse it as a string. so the controversial question is whether it should be used and if you use then change the type in the DTO itself or only in the validator

Reply:
@rogachevsn55 this has to be discussed, I think, whether we change DTO or not